### PR TITLE
[Naming] Apply ParamRenamer without conflict resolution for Naming set

### PR DESCRIPTION
**Pull request by dobryy**
Related to #4240

### CODE REVIEW

File: rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php [hunk 1]
```diff
@@ -17,40 +16,32 @@ abstract class AbstractExpectedNameResolver implements ExpectedNameResolverInter
...
-    ) {
+    public function __construct(NodeNameResolver $nodeNameResolver, NodeTypeResolver $nodeTypeResolver)
+    {
         $this->nodeNameResolver = $nodeNameResolver;
         $this->nodeTypeResolver = $nodeTypeResolver;
-        $this->propertyNaming = $propertyNaming;
     }
 
-    public function resolveIfNotYet(Property $property): ?string
+    public function resolveIfNotYet(Node $node): ?string
```
Comment (TomasVotruba) [inline rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php:30]: Here should be some Node type specification in `@param`

```suggestion
    /**
     * @name Property|Variable|...
     */
    public function resolveIfNotYet(Node $node): ?string
```

So the scope is narrow and it's easier to debug later. Now it can be anything, even `Switch_` :D 

Also these `Property|Variable|...` are [PHP 8-ready union types](https://wiki.php.net/rfc/union_types_v2). So we can upgrade Rector this year to:

```php
    public function resolveIfNotYet(Property|Variable Node $node): ?string
```

File: rules/naming/src/Rector/ClassMethod/RenameParamToMatchTypeRector.php [hunk 2]
```diff
@@ -99,8 +105,6 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        $this->hasChanged = false;
-
```
Comment (TomasVotruba) [inline rules/naming/src/Rector/ClassMethod/RenameParamToMatchTypeRector.php:103]: Why is this removed, when its still used bellow?

File: rules/naming/src/ValueObject/RenamePropertyValueObjectInterface.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+interface RenamePropertyValueObjectInterface extends RenameValueObjectInterface
+{
+    public function getClassLike(): ClassLike;
+
+    public function getClassLikeName(): string;
+
+    public function getProperty(): Property;
+
+    public function getPropertyProperty(): PropertyProperty;
+}
```
Comment (TomasVotruba) [inline rules/naming/src/ValueObject/RenamePropertyValueObjectInterface.php:20]: :+1:

**Review by TomasVotruba — COMMENTED** [review]
I've added few comment :)

File: rules/naming/src/Rector/ClassMethod/RenameParamToMatchTypeRector.php (hunk 2 above)
Comment (dobryy) [reply rules/naming/src/Rector/ClassMethod/RenameParamToMatchTypeRector.php:103]: Oooops, it was accidently removed, my bad..

File: rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php (hunk 1 above)
Comment (dobryy) [reply rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php:30]: For now it is just Property and Param.

**Comment by dobryy** [comment]
> I've added few comment :)

Ready :)

File: rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php (hunk 1 above)
Comment (TomasVotruba) [reply rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php:30]: These should be there.

Comment (TomasVotruba) [reply rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php:30]: I think also in the contract, to make the whole family of contract + implementer type aware

Comment (dobryy) [reply rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php:35]: Now, it should be okay.
Interface and abstract class have Property|Param.
Name resolvers have appropriate Node type(s) depending on the Node type(s) they handle.

Comment (TomasVotruba) [reply rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php:35]: Thank you
